sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/package.scala

```diff
@@ -41,6 +41,11 @@ package object analysis {
     def failAnalysis(msg: String): Nothing = {
       throw new AnalysisException(msg, t.origin.line, t.origin.startPosition)
     }
+
+    /** Fails the analysis at the point where a specific tree node was parsed. */
+    def failAnalysis(msg: String, cause: Throwable): Nothing = {
```

Comment:
I added a new method rather than using default parameter to maintain binary compatibly.

Reply:
BTW, I don't why the return type is `Nothing`. I just followed the above method.